src/jit/morph.cpp

```diff
@@ -15564,42 +15564,46 @@ void                Compiler::fgPromoteStructs()
     //
 
     lvaStructPromotionInfo structPromotionInfo;
+    bool tooManyLocals = false;
 
     for (unsigned lclNum = 0;
-        lclNum < startLvaCount;
-        lclNum++)
+         lclNum < startLvaCount;
+         lclNum++)
     {
+        bool canPromote = false;
```

Comment:
add a comment about this variable:
bool  canPromote = false;  // This gets set to true when we determine that it is legal to promote this variable